src/slurm_plugin/clustermgtd.py

```diff
@@ -749,142 +660,37 @@ def _update_static_nodes_in_replacement(self, slurm_nodes):
                 nodes_still_in_replacement.add(nodename)
         self._static_nodes_in_replacement = nodes_still_in_replacement
 
-    def _group_slurm_nodes(self, slurm_nodes, private_ip_to_instance_map):
+    def _group_slurm_nodes(self, slurm_nodes):
         """
         Group slurm nodes into different types.
 
         Check and return slurm nodes with unhealthy and healthy scheduler state, grouping unhealthy nodes
         by node type (static/dynamic).
         """
+        powering_down_nodes = []
         unhealthy_static_nodes = []
         unhealthy_dynamic_nodes = []
         bootstrap_failure_nodes = []
         healthy_nodes = []
         for node in slurm_nodes:
-            if not self._is_node_healthy(node, private_ip_to_instance_map):
-                if node.is_static:
+            if isinstance(node, DynamicNode) and node.is_powering_down_node_with_nodeaddr():
```

Comment:
How about having `is_powering_down` method for both `DynamicNode` and `StaticNode`? For `StaticNode` we can just return false, because it should never be in power_down.

Let's double check the code to avoid using `if isinstance(node, DynamicNode)`. This is the exact `if node.is_static` logic we are trying to remove by having distinct Objects for static and dynamic nodes.

Reply:
Ok, we just try to minimize the places where we check if a node is static or not. For any place we need to do the check, let's use `node.is_static` instead of `isinstance(node, DynamicNode)`, which is more robust. i.e. there is nothing preventing us accidentally initializing a static node as a DynamicNode